ls1intum/Artemis
# Change team mode on exercise import.

### Checklist

### Motivation and Context
See #2265: The team mode of exercises can not be changed during import, although there is no reason why it's not technically possible.

### Description
The check in the client that determined whether the team mode can be changed was faulty: It should only disable changes when editing the exercise, not when importing.
The server already supported this correctly, but I added JUnit tests to verify this.

### Steps for Testing
1. Create a (programming) exercise in individual mode
2. Import that exercise into a new exercise but change to team mode
3. Verify that everything works as expected: Teams can be created, students can be assigned to teams, etc.
4. Repeat steps 1-3 but swap individual & team mode (start with team mode and import to individual mode).
5. Make sure that you can not change the exercise mode when editing the exercises.
